CanDIG/katsu
# Sonchau/add testing

What's new:
- add factory-boy to requirements
- add trailing slash to ingest API to make it consistent with router
- update permission and authentication with is_admin mock
- remove extra id from nested serializer
- add factories class to generate mock data
- add tests for the required models and 1 optional model: program, donor, sample registration, specimen, primary diagnoses, treatment, follow up and chemotherapy.
- add codecov token to github action

How to test:
- pip install factory-boy
- run `tox` from the command line, should get all the tests pass
